tests/Unit/Controllers/Api/TraderControllerTest.php

```diff
@@ -72,4 +73,59 @@ public function testShowVoucherHistoryCompilesListOfPaymentHistory()
         $this->assertEquals($data[0]->vouchers[1]->recorded_on, $today);
         $this->assertEquals($data[0]->vouchers[2]->reimbursed_on, $today);
     }
+
+    /**
+     * Tests the email all voucher history API response.
+     */
+    public function testEmailVoucherHistoryAllDates()
+    {
+        // Sync the user with trader 1.
+        $this->user->traders()->sync([1]);
+        $this->actingAs($this->user, 'api')
+            ->json('POST', route('api.trader.voucher-history-email', 1), [
+                'submission_date' => null,
+            ])
+            ->assertStatus(202)
+            ->assertJson([
+                'message' => trans('api.messages.email_voucher_history')
+            ])
+        ;
+    }
+
+    /**
+     * Tests the email specific date voucher history API response.
+     */
+    public function testEmailVoucherHistorySpecificDate()
+    {
+        // Sync the user with trader 1.
+        $this->user->traders()->sync([1]);
+        // There should be some vouchers pended today.
+        $date = Carbon::now()->format('d-m-Y');
+        $this->actingAs($this->user, 'api')
+            ->json('POST', route('api.trader.voucher-history-email', 1), [
+                'submission_date' => null,
+            ])
+            ->assertStatus(202)
+            ->assertJson([
+                'message' => trans(
+                    'api.messages.email_voucher_history', [
```

Comment:
It's because you're not passing the date in the initial post. It's still null :)